llap-server/src/java/org/apache/hadoop/hive/llap/cache/SimpleAllocator.java

```diff
@@ -17,31 +17,17 @@
  */
 package org.apache.hadoop.hive.llap.cache;
 
-import java.lang.reflect.Field;
 import java.nio.ByteBuffer;
 
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.hive.common.io.Allocator;
 import org.apache.hadoop.hive.common.io.encoded.MemoryBuffer;
 import org.apache.hadoop.hive.conf.HiveConf;
 import org.apache.hadoop.hive.llap.io.api.impl.LlapIoImpl;
-
-import sun.misc.Cleaner;
+import org.apache.hive.common.util.CleanerUtil;
```

Comment:
why don't we use the `CleanerUtil` from hadoop? I don't think we should copy it

https://issues.apache.org/jira/browse/HADOOP-12760

Reply:
@kgyrtkirk I would prefer this too, but this class was made available in Hadoop 3.2.0 and Hive is on Hadoop 3.1.0

Reply:
I think we should upgrade to hadoop >=3.2.0 (in a separate ticket) prior to moving to jdk11...

Reply:
Other than saving a copy & paste on this one class, I'm not sure the value.  I don't think it should be a blocker on this.

Reply:
I think it's mandatory to upgrade to at least
https://issues.apache.org/jira/browse/HADOOP-15338 marks 3.3 for jdk11 support - I don't think we should try to run hive with hadoop 3.1 on jdk11...